common/src/main/java/com/microsoft/identity/common/exception/ClientException.java

```diff
@@ -145,6 +145,63 @@
      */
     public static final String DUPLICATE_COMMAND = "duplicate_command";
 
+    /**
+     * Emitted when the KeyStore generates a certificate that does not match the designated key size.
+     * Due to a bug in some versions of Android, keySizes may not be exactly as specified
+     * To generate a 2048-bit key, two primes of length 1024 are multiplied -- this product
+     * may be 2047 in length in some cases which causes Nimbus to crash. To avoid this,
+     * check the keysize prior to returning the generated KeyPair.
+     */
+    public static final String BAD_KEY_SIZE = "keystore_produced_invalid_cert";
+
+    /**
+     * Emitted when the requested crypto provider is unavailable in the device environment.
+     */
+    public static final String NO_SUCH_PROVIDER = "androidkeystore_unavailable";
```

Comment:
rename to `android_keystore_unavailable`

Reply:
sure - resolved in ca802575a925fcabdce7680d4c5c3045e6aa6d7e